### PR TITLE
[WIRES] small tweaks to wires class

### PR DESCRIPTION
**Pull request by mariaschuld**
**Context:**

Part of wires refactor. 

**Description of the Change:**

Adds ``at_index`` function to wires class, which returns the wire representation at a given index.

Adapts the class to deal correctly with strings as inputs. Formerly we would have

>>> Wires('h')
<Wires=['h', 'i']>

but now we have

>>> Wires('h')
<Wires=['hi']>

Added test for both.

### CODE REVIEW

File: pennylane/wires.py [hunk 1]
```diff
@@ -31,6 +31,10 @@ def _process(wires):
         # if input is already a Wires object, just return its wire tuple
         return wires.wire_tuple
 
+    elif isinstance(wires, (Number, str)):
+        # interpret as a single wire
+        return (wires,)
```
Comment (josh146) [inline pennylane/wires.py:36]: :+1: The drawbacks of Python treating strings as sequences of characters. Every now and again my code will fail and the output will be
```
['s', 'o', 'm', 'e', ' ', 's', 't', 'r', 'i', 'n', 'g']
```
and I'll know I've done it again.

File: pennylane/wires.py [hunk 2]
```diff
@@ -123,6 +123,23 @@ def tolist(self):
         """
         return list(self.wire_tuple)
 
+    def at_index(self, idx):
```
Comment (josh146) [inline pennylane/wires.py:126]: How does this differ from indexing the wire object directly?

File: pennylane/wires.py [hunk 3]
```diff
@@ -123,6 +123,23 @@ def tolist(self):
...
+        """Returns the wire at index ``idx``.
+
+        >>> w = Wires([0, 'q1', 16])
+        >>> w[1]
+        'q1'
+        >>> w[2]
```
Comment (josh146) [inline pennylane/wires.py:132]: Should this be `w.at_index(2)`? Same for `w[1]`.

File: pennylane/wires.py (hunk 2 above)
Comment (josh146) [reply pennylane/wires.py:126]: Is the difference that if you index a wire object directly, you get a wire back? E.g.,
```python
>>> w = Wires([0, 'q1', 16])
>>> w[1]
Wires(['q1'])
>>> w.at_index(1)
'q1'
```

File: pennylane/wires.py [hunk 4]
```diff
@@ -123,6 +123,23 @@ def tolist(self):
         """
         return list(self.wire_tuple)
 
+    def at_index(self, idx):
+        """Returns the wire at index ``idx``.
```
Comment (josh146) [inline pennylane/wires.py:127]: Tried to make it sound like one of the Python [list](https://docs.python.org/3/tutorial/datastructures.html) method docstrings 🤔 
```suggestion
        """Returns the wire representation at the given position in the wires object.
```

File: pennylane/wires.py (hunk 2 above)
Comment (josh146) [reply pennylane/wires.py:126]: If so, I'm wondering if it makes sense to name this something a bit more descriptive, that indicates you get the 'name' of the wire. Maybe `get_name`?
```python
>>> w = Wires([0, 'q1', 16])
>>> w[1]
Wires(['q1'])
>>> w.get_name(1)
'q1'
```

Comment (mariaschuld) [reply pennylane/wires.py:126]: Yes, exactly right. I was thinking of ``get_wire_representation`` but it is very long. ``get_name`` implies that it is a string...

File: pennylane/wires.py (hunk 3 above)
Comment (mariaschuld) [reply pennylane/wires.py:132]: AAAh, sorry!

Comment (mariaschuld) [reply pennylane/wires.py:132]:
```suggestion
        >>> w.at_index(2)
```

File: pennylane/wires.py [hunk 5]
```diff
@@ -123,6 +123,23 @@ def tolist(self):
...
+
+        >>> w = Wires([0, 'q1', 16])
+        >>> w[1]
```
Comment (mariaschuld) [inline pennylane/wires.py:130]:
```suggestion
        >>> w.at_index(1)
```

File: pennylane/wires.py (hunk 2 above)
Comment (mariaschuld) [reply pennylane/wires.py:126]: I really don't know. I thought ``at_index`` means "at the index i of my Wires object is something called 'q1' ". But if you think ``get_name`` is better we go with that. Or ``get_id``? We could start calling these things wire identifiers...

File: pennylane/wires.py [hunk 6]
```diff
@@ -123,6 +123,23 @@ def tolist(self):
         """
         return list(self.wire_tuple)
 
+    def get_label(self, idx):
+        """Returns the wire representation at the given position in the wires object.
```
Comment (josh146) [inline pennylane/wires.py:127]:
```suggestion
        """Returns the wire label at the given position in the wires object.
```

File: pennylane/wires.py [hunk 7]
```diff
@@ -123,6 +123,23 @@ def tolist(self):
...
+        >>> w.get_label(1)
+        'q1'
+        >>> w.get_label(2)
+        16
+
+        Args:
+            int: index of wire to return
+
+        Returns:
+            Number or str: representation of the wire
```
Comment (josh146) [inline pennylane/wires.py:139]:
```suggestion
            Number or str: label of the wire
```

File: tests/test_wires.py [hunk 8]
```diff
@@ -168,6 +168,15 @@ def test_convert_to_list(self):
         assert isinstance(list_, list)
         assert list_ == [4, 0, 1]
 
+    def test_at_index_method(self):
```
Comment (josh146) [inline tests/test_wires.py:171]:
```suggestion
    def test_get_label_method(self):
```

**Pull request merged**